### PR TITLE
Additional sign fixes

### PR DESCRIPTION
**Pull request by rcseacord**
mostly fixed problems related to sign comparison errors.

### CODE REVIEW

**Comment by osqueryer** [comment]
Can one of the osquery admins reply with "ok to test" to kick off a build...

**Comment by theopolis** [comment]
ok to test

File: CMakeLists.txt [hunk 1]
```diff
@@ -28,12 +28,7 @@ add_compile_options(
   -Wextra
   -Wstrict-aliasing
   -Wno-unused-parameter
-  -Wno-unused-result
   -Wno-missing-field-initializers
-  -Wno-sign-compare
-  -Wno-unused-local-typedef
```
Comment (theopolis) [inline CMakeLists.txt]: Currently, removing `-Wno-unused-local-typedef` is going to create pages and pages of warnings for boost (<=1.58) on OS X.

```
In file included from /Users/reed/git/github/osquery/osquery/config/update.cpp:11:
In file included from /Users/reed/git/github/osquery/include/osquery/config.h:25:
In file included from /Users/reed/git/github/osquery/include/osquery/flags.h:15:
In file included from /usr/local/include/boost/lexical_cast.hpp:30:
In file included from /usr/local/include/boost/range/iterator_range_core.hpp:42:
/usr/local/include/boost/range/algorithm/equal.hpp:174:13: warning: unused typedef 'boost_concept_check174' [-Wunused-local-typedef]
            BOOST_RANGE_CONCEPT_ASSERT(( SinglePassRangeConcept<const SinglePassRange1> ));
            ^
/usr/local/include/boost/range/concepts.hpp:102:45: note: expanded from macro 'BOOST_RANGE_CONCEPT_ASSERT'
    #define BOOST_RANGE_CONCEPT_ASSERT( x ) BOOST_CONCEPT_ASSERT( x )
                                            ^
/usr/local/include/boost/concept/assert.hpp:43:5: note: expanded from macro 'BOOST_CONCEPT_ASSERT'
    BOOST_CONCEPT_ASSERT_FN(void(*)ModelInParens)
    ^
/usr/local/include/boost/concept/detail/general.hpp:79:7: note: expanded from macro 'BOOST_CONCEPT_ASSERT_FN'
      BOOST_PP_CAT(boost_concept_check,__LINE__)             \
      ^
/usr/local/include/boost/preprocessor/cat.hpp:22:32: note: expanded from macro 'BOOST_PP_CAT'
#    define BOOST_PP_CAT(a, b) BOOST_PP_CAT_I(a, b)
                               ^
/usr/local/include/boost/preprocessor/cat.hpp:29:34: note: expanded from macro 'BOOST_PP_CAT_I'
#    define BOOST_PP_CAT_I(a, b) a ## b
                                 ^
<scratch space>:107:1: note: expanded from here
boost_concept_check174
```

File: osquery/extensions/CMakeLists.txt [hunk 2]
```diff
@@ -9,6 +9,7 @@ add_custom_command(
 
 add_compile_options(
   -Wno-macro-redefined
+  -fsanitize=address
```
Comment (theopolis) [inline osquery/extensions/CMakeLists.txt]: Remove this compile option.

**Comment by osqueryer** [comment]
Build is good! :+1:

**Comment by osqueryer** [comment]
Build is good! :+1:

**Comment by theopolis** [comment]
Looks good! Let me know if you're finished working on this branch/PR and we'll merge. We can clean up the remaining incorrect sign compares in follow up PRs.

**Comment by rcseacord** [comment]
I guess I am done.  I've moved on to some other testing.  The remaining incorrect sign compares I think would benefit from having core developers look at them.  In many cases, the idea solution is to change the type of function arguments, but this might not be feasible in all cases.  In some other cases, it might be adequate to cast to unsigned, or to convert to signed types.  In all cases, we have to be careful to make sure adequate range guarantees are in place, particularly with signed integers that can take on negative values.

**Comment by rcseacord** [comment]
hmm, found a bunch more sign problems i'm fixing now

**Comment by osqueryer** [comment]
Build is good! :+1:

**Pull request merged**